ScoutBot/Services/DatabaseService.cs

```diff
@@ -10,26 +12,75 @@ namespace ScoutBot.Services
     public class DatabaseService
     {
         /// <summary>
-        /// Adds the providerId to the database.
+        /// Adds a sheet to the sheets table.
         /// </summary>
-        /// <param name="providerId">The id from riot for generating tournaments.</param>
-        public static async Task PutProviderAsync(int providerId)
+        /// <param name="googleId">The id of the google sheet.</param>
+        /// <param name="name">The common name for the document.</param>
+        /// <returns></returns>
+        public static async Task<bool> AddSheetAsync(string googleId, string name)
         {
-            using (StatsContext db = new StatsContext())
+            using (ScoutContext db = new ScoutContext())
             {
-                await db.Providers.AddAsync(new Provider { ProviderID = providerId });
-                db.SaveChanges();
+                try
+                {
+                    await db.Sheets.AddAsync(new Sheets
+                    {
+                        GoogleId = googleId,
+                        Name = name
+                    });
+                    db.SaveChanges();
+                }
+                catch
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public static async Task<bool> AddSheetAccessAsync(ulong[] roles, string name)
+        {
+            using (ScoutContext db = new ScoutContext())
+            {
+                try
+                {
+                    Sheets sheet = await db.Sheets
+                                        .AsAsyncEnumerable()
+                                        .Where(s => s.Name == name)
+                                        .FirstAsync<Sheets>();
+
+                    foreach (ulong role in roles)
+                    {
+                        db.SheetAccess.Add(new SheetAccess
+                        {
+                            RoleId = role,
+                            SheetId = sheet.SheetId
+                        });
+                    }
+                    await db.SaveChangesAsync();
+                }
+                catch
```

Comment:
Catch the specific exception you're "trying" for. This way if another one arises, your program throws an error as it should